File: jee6RestScala/src/main/scala/org/timoponce/HelloBean.scala
package org.timoponce

import javax.faces.bean.{RequestScoped, ManagedBean}
import javax.inject.Inject
import javax.persistence.{PersistenceContext, EntityManager}
import javax.transaction.UserTransaction

@ManagedBean
@RequestScoped
class HelloBean {

  @Inject
  var persister: Persister=_

  def greetings(): String = {
    val car = new Car
    car.setName("car - " + System.currentTimeMillis())

    persister.persist(car)
    return "Scello World!"
  }
}

File: jee6Webapp/src/main/java/org/timoponce/HelloBean.java
package org.timoponce;

import javax.faces.bean.*;
import javax.persistence.*;
import java.util.*;
import javax.ejb.*;
import org.slf4j.*;

@ManagedBean
@RequestScoped
public class HelloBean {
	
  static final Logger LOG = LoggerFactory.getLogger(HelloBean.class);

  @EJB
  Persister persister;

  public String getGreetings(){
	//save a car
    Car car = new Car();
    car.setName("car - " + System.currentTimeMillis());
    persister.persist(car);	
	//list all cars
	StringBuilder buffer = new StringBuilder("[");	
	for(Car _car : persister.list(Car.class))
		buffer.append(_car.getName()).append(",");
	
	buffer.setLength( buffer.length() - 1 );		
	buffer.append("]");
    String msg = "Hello World! : " + buffer.toString();
	LOG.info(msg);
	return msg;
  }
}
